test/e2e/tls_util_test.go

```diff
@@ -49,55 +92,54 @@ func TestBothAppAndCATLSAssetsExist(t *testing.T) {
 			Namespace: namespace,
 		},
 	}
-
-	certName := "app-cert"
-	appSecret := &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: tlsutil.ToAppSecretName(crKind, crName, certName),
-		},
-	}
-	appSecret, err = f.KubeClient.CoreV1().Secrets(namespace).Create(appSecret)
+	actualAppSecret, actualCaConfigMap, actualCaSecret, err := cg.GenerateCert(mCR, nil, ccfg)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	caConfigMapAndSecretName := tlsutil.ToCASecretAndConfigMapName(crKind, crName)
-	caConfigMap := &v1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: caConfigMapAndSecretName,
-		},
+	if !reflect.DeepEqual(appSecret, actualAppSecret) {
+		t.Fatalf("expect %+v, but got %+v", appSecret, actualAppSecret)
 	}
-	caConfigMap, err = f.KubeClient.CoreV1().ConfigMaps(namespace).Create(caConfigMap)
-	if err != nil {
-		t.Fatal(err)
+	if !reflect.DeepEqual(caConfigMap, actualCaConfigMap) {
+		t.Fatalf("expect %+v, but got %+v", caConfigMap, actualCaConfigMap)
 	}
-
-	caSecret := &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: caConfigMapAndSecretName,
-		},
+	if !reflect.DeepEqual(caSecret, actualCaSecret) {
+		t.Fatalf("expect %+v, but got %+v", caSecret, actualCaSecret)
 	}
-	caSecret, err = f.KubeClient.CoreV1().Secrets(namespace).Create(caSecret)
+}
+
+// TestOnlyAppSecretExist tests a case where the application TLS asset exists but its correspoding CA asset doesn't. In this case, CertGenerator can't genereate a new CA because it won't verify the existing application TLS cert. Therefore, CertGenerator can't proceed and returns an error to the caller.
+func TestOnlyAppSecretExist(t *testing.T) {
+	f := framework.Global
+	ctx := f.NewTestCtx(t)
+	defer ctx.Cleanup(t)
+	namespace, err := ctx.GetNamespace()
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	cg := tlsutil.NewSDKCertGenerator(f.KubeClient)
-	ccfg := &tlsutil.CertConfig{
-		CertName: certName,
-	}
-	actualAppSecret, actualCaConfigMap, actualCaSecret, err := cg.GenerateCert(mCR, nil, ccfg)
+	_, err = f.KubeClient.CoreV1().Secrets(namespace).Create(appSecret)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	if !reflect.DeepEqual(appSecret, actualAppSecret) {
-		t.Fatalf("expect %+v, got %+v", appSecret, actualAppSecret)
+	cg := tlsutil.NewSDKCertGenerator(f.KubeClient)
+	// Use Pod as a dummy runtime object for the CR input of GenerateCert().
+	mCR := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind: crKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      crName,
+			Namespace: namespace,
+		},
 	}
-	if !reflect.DeepEqual(caConfigMap, actualCaConfigMap) {
-		t.Fatalf("expect %+v, got %+v", caConfigMap, actualCaConfigMap)
+	_, _, _, err = cg.GenerateCert(mCR, nil, ccfg)
+	if err == nil {
+		t.Fatal("expect error, but got none")
 	}
-	if !reflect.DeepEqual(caSecret, actualCaSecret) {
-		t.Fatalf("expect %+v, got %+v", caSecret, actualCaSecret)
+	expErrMsg := "ca secret and configMap are not found"
```

Comment:
If we're matching the exact error message then its best to define this a constant in `pkg/tlsutil` so we can reuse it here. Instead of defining it twice.